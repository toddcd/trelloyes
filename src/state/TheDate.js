import React from 'react';

class TheDate extends React.Component {
    constructor(props){
        super(props);
        this.state = { datetime: new Date() };

    }

    componentDidMount() {
        setInterval(() => {
            this.setState({ datetime: new Date() });

        }, 1000)
    }

    componentWillUnmount(){
        clearInterval(this.interval);
    }

    render() {

        return <div>{'The time is: '+this.state.datetime.toLocaleTimeString()}</div>
    }
}

export default TheDate