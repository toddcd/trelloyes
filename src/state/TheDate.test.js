import React from 'react';
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import TheDate from './TheDate'

describe(`TheDate Component`, () => {
    it.skip('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<TheDate />, div);
        ReactDOM.unmountComponentAtNode(div);
    })

    it('renders the UI as expected', () => {
        expect(
            renderer.create(<TheDate />).toJSON()
        ).toMatchSnapshot()
    })
})