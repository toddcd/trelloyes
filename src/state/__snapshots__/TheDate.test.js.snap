// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheDate Component renders the UI as expected 1`] = `
<div>
  The time is: 5:04:33 PM
</div>
`;
